### PR TITLE
add powershell script

### PR DESCRIPTION
**Pull request by gupta-kartik**

### CODE REVIEW

File: docs/deployment-examples/azure-deployments/deploying-to-azure-via-a-firewall/index.md [hunk 1]
```diff
@@ -6,6 +6,33 @@ position: 29
 
 All the Azure steps in Octopus are executed from the VM where the Octopus Server is running. So to able to successfully deploy to the Microsoft cloud, you need to make sure your Octopus Server can reach it through the network.
 
+To check if you are able to reach Microsoft cloud through your network, try running this script on the same machine and using an account with the same permissions as your Octopus Server.
```
Comment (wordlee) [inline docs/deployment-examples/azure-deployments/deploying-to-azure-via-a-firewall/index.md]: Suggestion to remove a few extra words:
```suggestion
To check you can reach Microsoft cloud through your network, run this script on the same machine using an account with the same permissions as your Octopus Server.
```

File: docs/deployment-examples/azure-deployments/deploying-to-azure-via-a-firewall/index.md [hunk 2]
```diff
@@ -6,6 +6,33 @@ position: 29
...
+
+:::info
+
+You might need to install Azure Powershell before running this script. More information on how to do this can be found at https://docs.microsoft.com/en-us/powershell/azure/install-az-ps?view=azps-2.5.0
```
Comment (wordlee) [inline docs/deployment-examples/azure-deployments/deploying-to-azure-via-a-firewall/index.md]: Updated the link formatting:
```suggestion
You might need to install Azure Powershell before running this script. For information on how to do, see [Install the Azure PowerShell module](https://docs.microsoft.com/en-us/powershell/azure/install-az-ps?view=azps-2.5.0).
```

File: docs/deployment-examples/azure-deployments/deploying-to-azure-via-a-firewall/index.md [hunk 3]
```diff
@@ -6,6 +6,38 @@ position: 29
...
+$AzureEnvironment = Get-AzEnvironment -Name $OctopusAzureEnvironment
+
+Connect-AzAccount -Credential $creds -TenantId $OctopusAzureADTenantId -SubscriptionId $OctopusAzureSubscriptionId -Environment $AzureEnvironment -ServicePrincipal
+
+Get-AzResourceGroup
+```
+
+You should get an output showing all the Azure Resource Groups you have access to
+
+![Screenshot of Azure Resource Groups](img.jpg)
```
Comment (wordlee) [inline docs/deployment-examples/azure-deployments/deploying-to-azure-via-a-firewall/index.md]:
```suggestion
![Screenshot of Azure Resource Groups](image.png)
```

File: docs/deployment-examples/azure-deployments/deploying-to-azure-via-a-firewall/index.md [hunk 4]
```diff
@@ -6,6 +6,38 @@ position: 29
...
+$securePassword = ConvertTo-SecureString $OctopusAzureADPassword -AsPlainText -Force
+$creds = New-Object System.Management.Automation.PSCredential ($OctopusAzureADClientId, $securePassword)
+$AzureEnvironment = Get-AzEnvironment -Name $OctopusAzureEnvironment
+
+Connect-AzAccount -Credential $creds -TenantId $OctopusAzureADTenantId -SubscriptionId $OctopusAzureSubscriptionId -Environment $AzureEnvironment -ServicePrincipal
+
+Get-AzResourceGroup
+```
+
+You should get an output showing all the Azure Resource Groups you have access to
```
Comment (wordlee) [inline docs/deployment-examples/azure-deployments/deploying-to-azure-via-a-firewall/index.md]:
```suggestion
You should get output showing all the Azure Resource Groups you have access to:
```

**Comment by gupta-kartik** [comment]
Thanks @wordlee. Changes made.

File: docs/deployment-examples/azure-deployments/deploying-to-azure-via-a-firewall/index.md [hunk 5]
```diff
@@ -6,6 +6,37 @@ position: 29
...
+$securePassword = ConvertTo-SecureString $OctopusAzureADPassword -AsPlainText -Force
+$creds = New-Object System.Management.Automation.PSCredential ($OctopusAzureADClientId, $securePassword)
+$AzureEnvironment = Get-AzEnvironment -Name $OctopusAzureEnvironment
+
+Connect-AzAccount -Credential $creds -TenantId $OctopusAzureADTenantId -SubscriptionId $OctopusAzureSubscriptionId -Environment $AzureEnvironment -ServicePrincipal
+
+Get-AzResourceGroup
+```
+
+You should see output showing all the Azure Resource Groups you have access to:
```
Comment (wordlee) [inline docs/deployment-examples/azure-deployments/deploying-to-azure-via-a-firewall/index.md]:
```suggestion
If everything is working as expected, you will see output showing all the Azure Resource Groups you have access to:
```

**Pull request merged**